tests/moe/test_b12x_w4a16_ep.py

```
@@ -0,0 +1,613 @@
+    with pytest.raises(ValueError, match=match):
+        _prepare_ep_expert_map(
+            torch.tensor(values, dtype=torch.int32),
+            num_local_experts=local_experts,
+            num_experts=len(values),
+        )
+
+
+def test_prepare_ep_expert_map_rejects_bad_tensor_forms():
+    from flashinfer.fused_moe.cute_dsl.b12x_moe import _prepare_ep_expert_map
+
+    with pytest.raises(TypeError, match="torch.int32"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)